vuepress/docs/next/docs/reference/bundle-upgrade.md

```diff
@@ -0,0 +1,136 @@
+
+### Add Micro Frontends & Microservices
+ 
+1. Add your micro frontend to the new bundle descriptor `entando.json`:
+``` sh
+ent bundle mfe add YOUR-MFE1
+```
+The docker-based bundle has a centralized bundle descriptor with bundle entities such as micro frontends, microservices, APIs, DBMS, and ID management specifications. 
+
+2. Copy the MFE files into `YOUR-BUNDLE-NAME/microfrontends` directory. For instance, for an MFE named YOUR-MFE1: 
+
+``` sh
+mv YOUR-ORIGIN-DIRECTORY/YOUR-MFE1/{.,}* microfrontends/YOUR-MFE1
+```
+Micro frontends and microservices are partitioned into their own directories in the docker-based bundles to streamline the CI/CD process.
+
+3. Add your microservice:
+``` sh
+ent bundle ms add YOUR-MS1
+```
+4. Copy the microservice YOUR-MS1 files to the new location:
+``` sh
+mv YOUR-ORIGIN-DIRECTORY/YOUR-MS1/{.,}* microservicess/YOUR-MS1
+```
+5. Repeat steps 1 through 4 for all micro frontends and microservices.
+
+### Add API Claims
+1. Now add an API claim to connect the `YOUR-MFE1` to a microsersive as needed. The connection information is added to the descriptor `entando.json`.
+```shell
+ent bundle api add YOUR-MFE1 YOUR-API --serviceName=YOUR-MS1 --serviceUrl=http://localhost:8081
+```
+
+2. Repeat the previous step for all APIs as needed.
+
+### Add Custom Commands to Test Locally
+
+1. For local development and testing, a custom command is needed in the `entando.json` to use a different port for each MFE as shown below: 
+``` 
+    "microfrontends": [
+        {
+            "name": "YOUR-MFE1",
+            "customElement": "YOUR-MFE1",
+            "stack": "react",
+            "type": "widget",
+            "group": "free",
+            "publicFolder": "public",
+            "commands": {
+                "run": "npm install && PORT=3000 npm start"
+            },
+            "apiClaims": [
+                {
+                    "name": "YOUR-API1",
+                    "type": "internal",
+                    "serviceName": "YOUR-MS1"
+                }
+            ]
+        },
+        {
+            "name": "YOUR-MFE2",
+            "customElement": "YOUR-MFE2",
+            "stack": "react",
+            "type": "widget",
+            "group": "free",
+            "publicFolder": "public",
+            "titles": {
+                "en": "YOUR-MFE2",
+                "it": "YOUR-MFE2"
+            },
+            "commands": {
+                "run": "npm install && PORT=3001 npm start"
+            },
+            "apiClaims": [
+                {
+                    "name": "YOUR-API2",
+                    "type": "internal",
+                    "serviceName": "YOUR-MS2"
+                }
+            ]
+        }
+    ]
+```
+Note that in production, all microservices must run on port 8081.
+
+### Add Auxiliary Services
+Copy the auxiliary services such as DBMS or Keycloak into the `svc` directory as needed and enable them for local testing. For a Keycloak service:
```

Comment:
"Copy **the** files **such as**" is kind of contradictory, because **the** is inclusive and specific while **such as** connotes optional. "Copy auxiliary services such as..." OR "Copy the auxiliary services (DBMS, Keycloak, etc.)..." is more precise